From 9f54989373f1fa3ead6ff1711c1d0d3d4f9e4dff Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <xiaowen.liu@freescale.com>
Date: Mon, 23 Nov 2015 12:09:16 -0500
Subject: [PATCH 609/635] MA-7121-2 enable camera hal v3 interface.

Fix video recording snapshot issue.
When do snapshot fast enough, there is NULL settings without previous set Frame.
Simplify mSettings assign when request->settings is not NULL.

Signed-off-by: Xiaowen Liu <xiaowen.liu@freescale.com>
---
 mx6/libcamera3/Camera.cpp | 16 +++-------------
 1 file changed, 3 insertions(+), 13 deletions(-)

diff --git a/mx6/libcamera3/Camera.cpp b/mx6/libcamera3/Camera.cpp
index c827bd3..9afd1c4 100644
--- a/mx6/libcamera3/Camera.cpp
+++ b/mx6/libcamera3/Camera.cpp
@@ -282,8 +282,6 @@ int32_t Camera::configureStreams(camera3_stream_configuration_t *stream_config)
     mStreams = newStreams;
     mNumStreams = stream_config->num_streams;
 
-    // Clear out last seen settings metadata
-    mSettings.clear();
     return 0;
 
 err_out:
@@ -393,18 +391,10 @@ int32_t Camera::processCaptureRequest(camera3_capture_request_t *request)
     ALOGV("%s:%d: Request Frame:%d Settings:%p", __func__, mId,
             request->frame_number, request->settings);
 
-    {
+    // NULL indicates use last settings
+    if (request->settings != NULL) {
         android::Mutex::Autolock al(mDeviceLock);
-        // NULL indicates use last settings
-        if (request->settings == NULL) {
-            if (mSettings == NULL || mSettings->isEmpty()) {
-                ALOGE("%s:%d: NULL settings without previous set Frame:%d Req:%p",
-                        __func__, mId, request->frame_number, request);
-                return -EINVAL;
-            }
-        } else {
-            mSettings = new Metadata(request->settings);
-        }
+        mSettings = new Metadata(request->settings);
     }
 
     if (request->input_buffer != NULL) {
-- 
1.8.0

